test/unit/org/apache/cassandra/gms/EndpointStateTest.java

```diff
@@ -199,4 +207,77 @@ public void testCCReleaseVersion() throws IOException
         EndpointState deserializedOld = EndpointState.serializer.deserialize(input, MessagingService.VERSION_3014);
         assertEquals(c3safeVersionString, deserializedOld.getApplicationState(ApplicationState.RELEASE_VERSION).value);
     }
+
+    @Test
+    public void testFilterOutgoingStates_30() throws UnknownHostException
+    {
+        Map<ApplicationState, VersionedValue> states = new EnumMap<>(ApplicationState.class);
+        states.put(ApplicationState.INTERNAL_ADDRESS_AND_PORT, valueFactory.internalAddressAndPort(InetAddressAndPort.getByName("10.0.0.1:7000")));
+        states.put(ApplicationState.NATIVE_ADDRESS_AND_PORT, valueFactory.nativeaddressAndPort(InetAddressAndPort.getByName("127.0.0.1:9042")));
+        Token token = DatabaseDescriptor.getPartitioner().getTokenFactory().fromString("1");
+        states.put(ApplicationState.STATUS_WITH_PORT, valueFactory.normal(Collections.singleton(token)));
+        states.put(ApplicationState.SSTABLE_VERSIONS, valueFactory.sstableVersions(Collections.singleton(VersionAndType.fromString("bti-na"))));
+        states.put(ApplicationState.INDEX_STATUS, valueFactory.indexStatus("test_index_status"));
+        states.put(ApplicationState.DC, valueFactory.datacenter("datacenter1"));
+        states.put(ApplicationState.DISK_USAGE, valueFactory.diskUsage("u:0.5"));
+
+        Map<ApplicationState, VersionedValue> filtered = new EnumMap<>(ApplicationState.class);
+        for (Map.Entry<ApplicationState, VersionedValue> entry : EndpointStateSerializer.filterOutgoingStates(states.entrySet(), MessagingService.VERSION_30))
+            filtered.put(entry.getKey(), entry.getValue());
+
+        assertEquals("10.0.0.1", filtered.get(ApplicationState.values()[7]).value);
```

Comment:
Can't we replace the ordinal by the constant?

Reply:
it's the ordinal of the enum in a different codebase (bdp).  i'll add a comment to make it clear.